MaxText/layers/initializers.py

```diff
@@ -39,3 +40,12 @@ def init_fn(key, shape, dtype, in_axis, out_axis):
     return fn(key, shape, dtype)
 
   return init_fn
+
+def variable_to_logically_partitioned(variable: nnx.VariableState):
```

Comment:
@bvandermoon I'm thinking of moving this function to `initializers.py` since its shared but maybe there is a better place?

Reply:
I'm ok with it here. Could also make a separate file like `variable_partitions.py` or something which would probably be better IMO